pkg/diag/validator/validator.go

```diff
@@ -0,0 +1,31 @@
+/*
+Copyright 2020 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validator
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type Status string
+
+type Validator interface {
+	// Run runs the validator and returns the list of resources with status.
+	Run(ctx context.Context, client kubernetes.Interface, ns string, opts metav1.ListOptions) ([]Resource, error)
```

Comment:
Could this be `Validate`  rather than `Run`?

Reply:
done!